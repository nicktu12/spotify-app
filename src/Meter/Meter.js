import React from 'react';
import PropTypes from 'prop-types';

export const Meter = (props) => {
  var {
    percent = 0,         // a number between 0 and 1, inclusive
    width = 100,         // the overall width
    height = 10,         // the overall height
    rounded = true,      // if true, use rounded corners
    color = "rgb(255,150,36)",   // the fill color
    animate = false,     // if true, animate when the percent changes
    label = null         // a label to describe the contents (for accessibility)
  } = props;

  var round =  rounded ? Math.ceil(height / 2) : 0;
  var otherWidth =  percent ? 
    Math.max(height, width * Math.min(percent, 1)) :
    0;
  var style = animate ? { "transition": "width 500ms, fill 250ms" } : null;

  return (
    <svg width={width} height={height} aria-label={label}>
      <rect 
        width={width} 
        height={height} 
        fill="rgb(22,21,21)" 
        rx={round} 
        ry={round}
      />
      <rect 
        width={otherWidth} 
        height={height} 
        fill={color} 
        rx={round} 
        ry={round} 
        style={style}
      />
    </svg>
  );
};

Meter.propTypes = {
  percent: PropTypes.number,
  width: PropTypes.number,
  height: PropTypes.number,
  rounded: PropTypes.bool,
  color: PropTypes.string,
  animate: PropTypes.bool,
  label: PropTypes.number,
};
