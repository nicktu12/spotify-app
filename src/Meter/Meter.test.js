import React from 'react';
import { shallow } from 'enzyme';
import { Meter } from './Meter';

describe('Meter test', ()=>{
  it('should match snapshot render', ()=>{
    const wrapper = shallow(<Meter />);

    expect(wrapper).toMatchSnapshot();
  });
});
