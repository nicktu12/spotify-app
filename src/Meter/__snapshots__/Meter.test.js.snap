// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Meter test should match snapshot render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Meter />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "aria-label": null,
      "children": Array [
        <rect
          fill="rgb(62,61,61)"
          height={10}
          rx={5}
          ry={5}
          width={100}
        />,
        <rect
          fill="rgb(255,150,36)"
          height={10}
          rx={5}
          ry={5}
          style={null}
          width={0}
        />,
      ],
      "height": 10,
      "width": 100,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "fill": "rgb(62,61,61)",
          "height": 10,
          "rx": 5,
          "ry": 5,
          "width": 100,
        },
        "ref": null,
        "rendered": null,
        "type": "rect",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "fill": "rgb(255,150,36)",
          "height": 10,
          "rx": 5,
          "ry": 5,
          "style": null,
          "width": 0,
        },
        "ref": null,
        "rendered": null,
        "type": "rect",
      },
    ],
    "type": "svg",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "aria-label": null,
        "children": Array [
          <rect
            fill="rgb(62,61,61)"
            height={10}
            rx={5}
            ry={5}
            width={100}
          />,
          <rect
            fill="rgb(255,150,36)"
            height={10}
            rx={5}
            ry={5}
            style={null}
            width={0}
          />,
        ],
        "height": 10,
        "width": 100,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "fill": "rgb(62,61,61)",
            "height": 10,
            "rx": 5,
            "ry": 5,
            "width": 100,
          },
          "ref": null,
          "rendered": null,
          "type": "rect",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "fill": "rgb(255,150,36)",
            "height": 10,
            "rx": 5,
            "ry": 5,
            "style": null,
            "width": 0,
          },
          "ref": null,
          "rendered": null,
          "type": "rect",
        },
      ],
      "type": "svg",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
